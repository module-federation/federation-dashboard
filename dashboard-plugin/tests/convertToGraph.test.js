const fs = require("fs");
const convertToGraph = require("../convertToGraph");
const mockDataDir = fs.readdirSync(`${__dirname}/mock-data`);

describe("should convert Plugin data to graph", () => {
  const dataListQueue = [];
  let currentData;

  beforeAll(() => {
    for (let file of mockDataDir) {
      currentData = require(`${__dirname}/mock-data/${file}`);
      dataListQueue.push(currentData);
    }
  });

  test("should convert data to graph", () => {
    const rawData = dataListQueue.shift();
    const graph = convertToGraph(rawData);

    expect(graph.consumes instanceof Array).toBe(true);
    expect(graph.dependencies instanceof Array).toBe(true);
    expect(graph.devDependencies instanceof Array).toBe(true);
    expect(typeof graph.id === "string").toBe(true);
    expect(graph.modules instanceof Array).toBe(true);
    expect(typeof graph.name === "string").toBe(true);
    expect(graph.overrides instanceof Array).toBe(true);
    expect(graph.optionalDependencies instanceof Array).toBe(true);
    expect(typeof graph.remote === "string").toBe(true);
    expect(graph).toMatchSnapshot();
  });
});
