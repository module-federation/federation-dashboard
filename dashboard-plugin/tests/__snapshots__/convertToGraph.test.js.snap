// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should convert Plugin data to graph should convert data to graph 1`] = `
Object {
  "consumes": Array [],
  "dependencies": Array [],
  "devDependencies": Array [
    Object {
      "name": "@babel/core",
      "version": "7.9",
    },
    Object {
      "name": "@babel/preset-react",
      "version": "7.9",
    },
    Object {
      "name": "@module-federation/dashboard-plugin",
      "version": "1.0",
    },
    Object {
      "name": "babel-loader",
      "version": "8.1",
    },
    Object {
      "name": "serve",
      "version": "11.3",
    },
    Object {
      "name": "webpack",
      "version": "5.0.0-beta",
    },
    Object {
      "name": "webpack-cli",
      "version": "3.3",
    },
    Object {
      "name": "webpack-dev-server",
      "version": "3.11",
    },
    Object {
      "name": "@module-federation/automatic-vendor-federation",
      "version": "1.0",
    },
  ],
  "id": "utils",
  "modules": Array [
    Object {
      "applicationID": "utils",
      "file": "./src/analytics",
      "id": "utils:analytics",
      "name": "analytics",
      "requires": Array [],
    },
  ],
  "name": "utils",
  "optionalDependencies": Array [],
  "overrides": Array [],
  "remote": "http://localhost:3005/remoteEntry.js",
}
`;
